lib/ansible/modules/web_infrastructure/pmp.py

```diff
@@ -0,0 +1,262 @@
+        resources = json.loads(open_url(url, validate_certs=self.validate_certs, use_proxy=self.use_proxy).read())
+        for res in resources['operation']['Details']:
+            if res['RESOURCE NAME'] == name:
+                return res
+        raise ValueError('RESOURCE not found!')
+
+    def get_password(self, server, username):
+        resource = self.get_resource_by_name(server)
+        self.data['resourceId'] = resource['RESOURCE ID']
+        account = self.get_account_resource_by_name(username)
+        self.data['accountId'] = account['ACCOUNT ID']
+        return self.get_account_password()
+
+    def delete_resource(self, resourcename):
+        try:
+            resource = self.get_resource_by_name(resourcename)
+            self.data['resourceId'] = resource['RESOURCE ID']
+        except:
+            return {'changed': False, 'msg': 'Resource doesn\'t exists'}
+
+        url = self.format('https://{HOST}:{PORT}/restapi/json/v1/resources/{resourceId}?AUTHTOKEN={TOKEN}')
+        response = json.loads(
+            open_url(url, method="DELETE", validate_certs=self.validate_certs, use_proxy=self.use_proxy).read())
+        status = response['operation']['result']['status']
+        if status == 'Failed':
+            raise ValueError(response['operation']['result']['message'])
+        return {'changed': True, 'msg': 'Succesfully deleted resource'}
+
+    def create_resource(self, resourcename, accountname, password, ownername, resource_type='Linux', group=''):
+        data = {
+            "operation": {
+                "Details": {
+                    "RESOURCENAME": resourcename,
+                    "ACCOUNTNAME": accountname,
+                    "RESOURCETYPE": resource_type,
+                    "PASSWORD": password,
+                    "OWNERNAME": ownername,
+                }
+            }
+        }
+        if group:
+            data["operation"]["Details"]["RESOURCEGROUPNAME"] = group
+        headers = {
+            'Content-Type': 'text/json',
+        }  # admin test
+
+        url = self.format('https://{HOST}:{PORT}/restapi/json/v1/resources?AUTHTOKEN={TOKEN}')
+        data = json.dumps(data, sort_keys=False)
+        params = urllib.urlencode({'INPUT_DATA': data})
+        r = open_url(url, method="POST", headers=headers, data=params, validate_certs=self.validate_certs,
+                     use_proxy=self.use_proxy)
+        res = json.loads(r.read())['operation']['result']
+        status = res['status']
+        if status == 'Failed':
+            raise ValueError(res['message'])
+        return {'changed': True, 'msg': 'Succesfully created resource'}
+
+
+def main():
+    global module
+    module = AnsibleModule(
+        argument_spec=dict(
+            uri=dict(required=True),
+            port=dict(required=False, default="7272"),
+            state=dict(choices=['present', 'absent'], required=True),
```

Comment:
In most modules, `state` is optional and defaults to `present`.  It would be easier for users if this module stuck to that convention.